internal/core/src/common/QueryResult.h

```diff
@@ -31,6 +32,106 @@
 #include "knowhere/index_node.h"
 
 namespace milvus {
+
+struct OffsetDisPair {
+    std::pair<int64_t, float> off_dis_;
+    int iterator_idx_;
+
+ public:
+    OffsetDisPair(std::pair<int64_t, float> off_dis, int iter_idx)
+        : off_dis_(off_dis), iterator_idx_(iter_idx) {
+    }
+};
+
+const std::pair<int64_t, float> EmptyOffsetPair = std::make_pair(-1, -1.0);
+
+struct OffsetDisPairComparator {
+    bool
+    operator()(const std::shared_ptr<OffsetDisPair>& left,
+               const std::shared_ptr<OffsetDisPair>& right) const {
+        if (left->off_dis_.second != right->off_dis_.second) {
+            return left->off_dis_.second < right->off_dis_.second;
+        }
+        return left->off_dis_.first < right->off_dis_.first;
+    }
+};
+struct VectorIterator {
+ public:
+    VectorIterator(int chunk_count, int64_t chunk_rows = -1)
+        : chunk_rows_(chunk_rows) {
+        iterators_.reserve(chunk_count);
+    }
+
+    std::pair<int64_t, float>
```

Comment:
prefer to use `std::optional<T>` then we can remove the magic value `EmptyOffsetPair`

Reply:
changed